assets/js/base/components/cart-checkout/product-name/style.scss

```diff
@@ -1,4 +1,6 @@
 .wc-block-product-name {
 	@include font-size(16);
 	@include wrapBreakWord();
+	display: block;
+	width: max-content;
```

Comment:
This CSS property doesn't work on IE11, but that's only to make the clickable area or the link smaller, so I think it can be comprised inside graceful degradation for old browsers.